zap/src/main/java/org/parosproxy/paros/core/scanner/AbstractAppParamPlugin.java

```diff
@@ -218,4 +219,15 @@ protected String setParameter(HttpMessage message, String param, String value) {
     protected String setEscapedParameter(HttpMessage message, String param, String value) {
         return variant.setEscapedParameter(message, originalPair, param, value);
     }
+
+    /**
+     * Sets the parameters into the given {@code message}.
+     *
+     * @param message the message that will be changed
+     * @param appParams list of the parameters
+     * @return the parameter values which are added to the HttpMesage
+     */
+    protected List<String> setParameters(HttpMessage message, List<AppParameter> appParams) {
```

Comment:
Should also provide a method that returns the builder and both methods should mention each other, to make it clear how the API is supposed to be used.

Reply:
ok, mention meaning java doc mention?

Reply:
Right.